### PR TITLE
Fix/php7.4

### PR DESCRIPTION
**Pull request by jbout**
Fix PHP 7.4 serialization by only storing the id in the session, not the resource

### CODE REVIEW

File: models/classes/TaoLtiSession.php [hunk 1]
```diff
@@ -89,12 +87,11 @@ public function getLaunchData()
...
-        if (is_null($this->ltiLink)) {
+        if (is_null($this->ltiLinkId)) {
             $service = $this->getServiceLocator()->get(LinkService::SERVICE_ID);
             $consumer = $this->getLaunchData()->getLtiConsumer();
-            $linkId = $service->getLinkId($consumer->getUri(), $this->getLaunchData()->getResourceLinkID());
-            $this->ltiLink = new core_kernel_classes_Resource($linkId);
+            $this->ltiLinkId = $service->getLinkId($consumer->getUri(), $this->getLaunchData()->getResourceLinkID());
         }
-        return $this->ltiLink;
+        return new core_kernel_classes_Resource($this->ltiLinkId);
```
Comment (SergiiTao) [inline models/classes/TaoLtiSession.php:95]: why do we need to have a new instance of the resource on every call?

Comment (SergiiTao) [reply models/classes/TaoLtiSession.php:95]: can we use `clone` instead of `new` here? it should be a bit faster

Comment (julien-sebire) [reply models/classes/TaoLtiSession.php:95]: This is about deserialization. It's better to serialize/deserialize an id thatn a resource object.

Comment (julien-sebire) [reply models/classes/TaoLtiSession.php:95]: @jbout Do we have existing serialized data that we've to take care about when changing the way we deserialize?

**Review by julien-sebire — DISMISSED** [review]
@jbout Do we have existing serialized data that we've to take care of when changing the way we deserialize? I mean, some already serialized resources? Should we check the type of the serialized ltiLink to check whether it's an id or a resource?
Also, do we have a way to put some testing about this change?

File: models/classes/TaoLtiSession.php (hunk 1 above)
Comment (SergiiTao) [reply models/classes/TaoLtiSession.php:95]: I don't think it makes a big difference if we do not serialize `common_session_DefaultSession` in the 'proper' way

**Comment by jbout** [comment]
> @jbout Do we have existing serialized data that we've to take care of when changing the way we deserialize? I mean, some already serialized resources? Should we check the type of the serialized ltiLink to check whether it's an id or a resource?

```new core_kernel_classes_Resource($mixed)``` either supports an id OR a resource as a parameter, so the fallback is already in the constructor

**Comment by jbout** [comment]
Providing a test case is unfortunately not possible without doing major changes to our session and user serialization

**Comment by julien-sebire** [comment]
> Providing a test case is unfortunately not possible without doing major changes to our session and user serialization
Two ways of looking at the choice to do then: either you make so the machine can test (that requires unfortunate work to be done) or you make so the humans have to do the machine work... ;-)

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
lgtm, 
tested with docker image and php version 
PHP 7.4.9 (cli) (built: Sep  1 2020 02:33:08) ( NTS )
Copyright (c) The PHP Group

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request closed without merging**